client/client.go

```diff
@@ -307,6 +307,26 @@ func (w *WrapperV2) DeleteCluster(clusterID string, p *AuxiliaryParams) (*cluste
 	return response, nil
 }
 
+// GetCluster fetches details on a clsuter using the latest client.
+func (w *WrapperV2) GetCluster(clusterID string, p *AuxiliaryParams) (*clusters.GetClusterOK, error) {
+	if w == nil {
```

Comment:
This feels weird. My gut says you shouldn't be able to get into a point where you're calling against nil

Reply:
Ideally no. But it did happen to me when the client wasn't initialized properly. So how could I prevent it in a more elegant way?

Reply:
LoL https://play.golang.org/p/GoslHrX9Vtm
I didn't know that's possible

Reply:
It makes me worried - it feels like you have some cases where you're not constructing the client correctly. I'd not have this check, and solve it there.

Reply:
> I didn't know that's possible

I did the exact same thing (including the lol message)

There is also a common good practice that says, avoid pointer whenever possible and only use pointer receiving method when you actually need to change some value in the object.
So here you can change from `func (w *WrapperV2) GetCluster` to `func (w WrapperV2) GetCluster`

Reply:
Where is the `WrapperV2` initialized? Why it can be `nil` and we should handle that?